### ISSUE TITLE
Initialisation slow?

### ISSUE THREAD

**Issue by Ponsen** [muaz-khan/DetectRTC#6]
Hi!

First of all: Great tool! Unfortunately I cannot get it to work.

Szenario:
Page opens. I'm calling DetectRTC.load() in my HTML right in the header after including the script and then initializing all WebRTC related stuff when $.ready is called. But when I want to use the variables like "hasWebcam" or "hasMicrophone" they're telling me that nothing was detected.
Strangest thing: when I'm debugging your lib to see what's going on, all variables are getting initialized (correctly!) and when I reach the point where it usually fails: All variables are available. So I'm heaving some kind of asynchronous issue.

In short:
without breakpoints -> not working
with -> working

Can you help me?

Thanks a lot in advance!

**Comment by muaz-khan** [comment]
Both demos ( [HTTPs](https://www.webrtc-experiment.com/DetectRTC/) and [HTTP](http://jsfiddle.net/opfqmhrv/) ) seems working for me, using Chrome.

Can you setup a demo somewhere on jsfiddle or cloud9? BTW, are you using iframes or?

Did you check Chrome address bar's top-right corner to see if audio/video devices are blocked?

**Comment by Ponsen** [comment]
I studied you code a bit and realized you're initializing a RTC connecting to evaluate the variables. This kinda came in conflict with my initialization which happens right on page load. So that was on of the problems. The other problem is javascript itself, cause it's running asynchronously. So it's on my side to correct the lifecycle.

A workaround I'm doing right now is to set the variables on a landing page, save them in a cookie, and just use them later when needed. If the user doesn't have those cookies, I'm sending him back to the landingpage.

Anyway. Thank you a lot for your fast reply and have a nice day!

**Comment by muaz-khan** [comment]
Please check https://github.com/muaz-khan/DetectRTC/issues/35#issuecomment-285848241
